server/src/analysis/test/resolveReference.test.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * This file is part of OpenModelica.
+ *
+ * Copyright (c) 1998-2024, Open Source Modelica Consortium (OSMC),
+ * c/o Linköpings universitet, Department of Computer and Information Science,
+ * SE-58183 Linköping, Sweden.
+ *
+ * All rights reserved.
+ *
+ * THIS PROGRAM IS PROVIDED UNDER THE TERMS OF AGPL VERSION 3 LICENSE OR
+ * THIS OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.8.
+ * ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+ * RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GNU AGPL
+ * VERSION 3, ACCORDING TO RECIPIENTS CHOICE.
+ *
+ * The OpenModelica software and the OSMC (Open Source Modelica Consortium)
+ * Public License (OSMC-PL) are obtained from OSMC, either from the above
+ * address, from the URLs:
+ * http://www.openmodelica.org or
+ * https://github.com/OpenModelica/ or
+ * http://www.ida.liu.se/projects/OpenModelica,
+ * and in the OpenModelica distribution.
+ *
+ * GNU AGPL version 3 is obtained from:
+ * https://www.gnu.org/licenses/licenses.html#GPL
+ *
+ * This program is distributed WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+ * IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
+ *
+ * See the full OSMC Public License conditions for more details.
+ *
+ */
+
+import Parser from 'web-tree-sitter';
+import assert from 'node:assert/strict';
+import path from 'node:path';
+import { ModelicaProject, ModelicaLibrary, ModelicaDocument } from '../../project';
+import { initializeParser } from '../../parser';
+import resolveReference from '../resolveReference';
+import {
+  UnresolvedAbsoluteReference,
+  UnresolvedRelativeReference,
+  ResolvedReference,
+} from '../reference';
+import * as TreeSitterUtil from '../../util/tree-sitter';
+
+const TEST_LIBRARY_PATH = path.join(__dirname, 'TestLibrary');
```

Comment:
Maybe also test a directory named `TestLibrary 1.0.0` (with a space in the name). Looks like that is an issue.